### PR TITLE
Simplify setting connections and other issues

### PR DESCRIPTION
**Pull request by maciekbanas**
- simplify setting connections and deprecate previous method, together with new tests
- refactor map iterators for get repos method (move iterations over repo owners from privates to publics),
- create CI

### CODE REVIEW

File: README.Rmd [hunk 1]
```diff
@@ -15,69 +15,128 @@ knitr::opts_chunk$set(
...
 
-my_gitlab <- GitLabClient$new(
-  groups = c("my_group", "my_second_group"),
-  rest_api_url = "your_api_url.com",
-  token = Sys.getenv("GITLAB_PAT)
-)
+my_gitstats <- GitStats$new()
+
+my_gitstats$set_connection(api_url = "https://api.github.com",
+                           token = Sys.getenv("GITHUB_PAT"),
```
Comment (Cotau) [inline README.Rmd:48]: you could add a short instruction how to setup that variable, also I ran the code without it in renviron which didn't raise any error, you could consider checking connection on the fly and raising issue if that didn't work before user will continue with the package and will get errors everywhere else

File: .Rprofile [hunk 2]
```diff
@@ -0,0 +1,4 @@
+if (Sys.getenv("USE_RENV", unset = "TRUE") == "TRUE") {
```
Comment (Cotau) [inline .Rprofile:1]: is this some form of high level check of environmental variable? 
Was it initialised automatically by creating renv? I think that for now you should jest setup renv and update it with new package added. I'm not quite sure if complicated workflow with package checks is the best idea at this point

File: README.Rmd [hunk 3]
```diff
@@ -15,69 +15,128 @@ knitr::opts_chunk$set(
...
-  token = Sys.getenv("GITHUB_PAT")
-)
+library(GitStats)
 
-my_gitlab <- GitLabClient$new(
-  groups = c("my_group", "my_second_group"),
-  rest_api_url = "your_api_url.com",
-  token = Sys.getenv("GITLAB_PAT)
-)
+my_gitstats <- GitStats$new()
```
Comment (Cotau) [inline README.Rmd:45]: I think you could build a vignette with example usage, you could put some examples in there  and in general readme only focus on basics

File: inst/example_workflow.R [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  owners_groups = c("openpharma", "r-world-devs")
+)
+
+git_stats
+
+# examples for getting repos
+
+git_stats$get_repos_by_owner_or_group()
+
+git_stats$get_repos_by_codephrase("Dashboard")
```
Comment (Cotau) [inline inst/example_workflow.R:17]: running this line raises an error on my side:
`Error in `resp_check_status()`:
! HTTP 403 Forbidden.
• You have exceeded a secondary rate limit. Please wait a few minutes before you try again.
• See docs at <https://docs.github.com/en/free-pro-team@latest/rest/overview/resources-in-the-rest-api#secondary-rate-limits>`

File: inst/example_workflow.R [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+                   "kalimu",
+                   "marcinkowskak",
+                   "Cotau",
+                   "maciekbanas")
+
+git_stats$get_repos_by_team("RWD")
+
+# you can plot repos sorted by last activity
+
+git_stats$plot_repos()
```
Comment (Cotau) [inline inst/example_workflow.R:36]: it rises warning, though plots are generated and look nice:
`Warning messages:
1: In RColorBrewer::brewer.pal(N, "Set2") :
  minimal value for n is 3, returning requested palette with 3 different levels

2: In RColorBrewer::brewer.pal(N, "Set2") :
  minimal value for n is 3, returning requested palette with 3 different levels`

File: inst/example_workflow.R [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+# examples for getting commits
+# still based on GraphQL for GitHub with poor performance
+
+git_stats$get_commits_by_team("RWD", date_from = "2022-01-01")$plot_commits(stats_by = "day")
+
+# may occur fatal Bad Gateway error - one of reasons to deprecate GraphQL in GitHub class
+
+git_stats$get_commits_by_team("RWD", date_from = "2020-01-01")$plot_commits(stats_by = "month")
+
+git_stats$get_commits_by_owner_or_group(date_from = "2022-01-01")$plot_commits(stats_by = "month")
```
Comment (Cotau) [inline inst/example_workflow.R:55]: Bad Gateway error as noted:
`Bad Gateway (HTTP 502)
Github (openpharma): pulling commits from 37 repositories.                                  
Github (r-world-devs): pulling commits from 5 repositories.
Error in `[.data.table`(x, i, , ) : 
  Column 5 ['authors'] is a data.frame or data.table; malformed data.table.
In addition: Warning messages:
1: In setDT(ans) :
  Some columns are a multi-column type (such as a matrix column): [5]. setDT will retain these columns as-is but subsequent operations like grouping and joining may fail. Please consider as.data.table() instead which will create a new column for each embedded column.
2: In setDT(ans) :
  Some columns are a multi-column type (such as a matrix column): [5]. setDT will retain these columns as-is but subsequent operations like grouping and joining may fail. Please consider as.data.table() instead which will create a new column for each embedded column.
3: In setDT(ans) :
  Some columns are a multi-column type (such as a matrix column): [5]. setDT will retain these columns as-is but subsequent operations like grouping and joining may fail. Please consider as.data.table() instead which will create a new column for each embedded column.
4: In setDT(ans) :`

File: renv/.gitignore [hunk 7]
```diff
@@ -0,0 +1,6 @@
+library/
```
Comment (Cotau) [inline renv/.gitignore:1]: you got one gitignore in package directory and one here, how do they differ from eachother?

File: renv/activate.R [hunk 8]
```diff
@@ -0,0 +1,933 @@
+
+local({
```
Comment (Cotau) [inline renv/activate.R:2]: I am not familiar with the setup workflow, to be discussed

**Review by Cotau — CHANGES_REQUESTED** [review]
I didn't looked too deep into code, was trying to get the view of how the package is constructed. I've left several comments and questions to be discussed

File: LICENSE [hunk 9]
```diff
@@ -1,21 +1,2 @@
-MIT License
```
[the remark this answers is not in the document]
Comment (maciekbanas) [reply LICENSE:1]: Yes, it was agreed upon MIT license.

File: inst/example_workflow.R (hunk 4 above)
Comment (maciekbanas) [reply inst/example_workflow.R:17]: Yes, I need to fix searching requests, I set up an issue for that in #32

File: .Rprofile (hunk 2 above)
Comment (maciekbanas) [reply .Rprofile:1]: Basically, installing renv broke CI jobs (CDM checks and test coverage) - this code is to overcome it and fix checks, it is based on @krystian8207 's commit activity on DataFakeR.

File: .Renviron [hunk 10]
```diff
@@ -0,0 +1 @@
+_R_CHECK_SYSTEM_CLOCK_=0
```
[the remark this answers is not in the document]
Comment (maciekbanas) [reply .Renviron:1]: I will add a comment in a file to explain what it is used for.

File: .github/workflows/check-standard.yaml [hunk 11]
```diff
@@ -0,0 +1,49 @@
+# Workflow derived from https://github.com/r-lib/actions/tree/v2/examples
```
[the remark this answers is not in the document]
Comment (maciekbanas) [reply .github/workflows/check-standard.yaml:1]: I created an issue to add spelling and styling checks #34

File: README.Rmd (hunk 1 above)
Comment (maciekbanas) [reply README.Rmd:48]: I added a short info in comment next to token. I will prepare warning for the 'empty token' case. #36

File: inst/example_workflow.R (hunk 5 above)
Comment (maciekbanas) [reply inst/example_workflow.R:48]: I created an issue for that #35

File: renv/.gitignore (hunk 7 above)
Comment (maciekbanas) [reply renv/.gitignore:1]: Not sure if it is correct, but as I search through repos that is quite common, that .gitignore files are in root as well as in renv and .github folders. As I read, keeping .gitignore in subfolder makes sense in that way all the file-rules apply to the root directory of .gitignore file. https://stackoverflow.com/questions/53208235/multiple-gitignore-in-subfolders
Can we keep it that way or do you have other view?

File: README.Rmd (hunk 3 above)
Comment (maciekbanas) [reply README.Rmd:45]: I added an issue to build vignette #37

**Review by Cotau — APPROVED** [review]
lgtm!

**Pull request merged**